opentasks/src/main/java/org/dmfs/tasks/utils/DateFormatter.java

```diff
@@ -315,4 +339,19 @@ private String formatNonAllDay(Time date, Time now, DateFormatContext dateContex
     {
         return DateUtils.formatDateTime(mContext, date.toMillis(false), dateContext.getDateUtilsFlags(now, date));
     }
+
+
+    /**
+     * {@link Time} will eventually be replaced with {@link DateTime} in the project.
+     * This conversion function is only needed in the transition period.
+     */
+    @VisibleForTesting
+    Time toTime(DateTime dateTime)
+    {
+        Time time = new Time();
+        time.set(dateTime.getTimestamp());
+        time.timezone = dateTime.getTimeZone().getID();
```

Comment:
This will crash if the `dateTime` has no time zone (I didn't have `Optional` back then) like it's the case for all-day tasks.

Reply:
I spent more time now with this conversion, updated the conversion code and the test as well (which now passes). I assumed that we don't support non-all-day floating `DateTime` because we can't represent that with `Time`. So the conversion throws now in that case.

It is also not clear what is best way to set the alll-day  flag on `Time`, directly or with a `set(year, month, day)` method, I left a TODO to check back on that. 

Let me know if this `DateFormatter` update should be submitted with a separate ticket and pull request.